File: 01_basics/01_07_armstrong/armstrong.js
const armstrong = (n) => {
    let temp = n;
    let count = 0
    let sum = 0;

    while (temp > 0) {
        count++;
        temp = Math.floor(temp / 10);
    }

    temp = n;
    while (temp > 0) {
        const digit = temp % 10;
        sum += Math.pow(digit, count);
        temp = Math.floor(temp / 10);
    }

    return sum == n;
}

const num = 153;
console.log(`Is ${num} and Armstrong number: ${armstrong(num)}`);

File: 01_basics/01_07_armstrong/Armstrong.java
public class Armstrong {
    public static void main(String[] args) {
        int num = 153;

        System.out.println(String.format("Is %d an Armstrong number: %b", num, armstrong(num)));
    }

    private static boolean armstrong(int num) {
        int temp = num;
        int count = 0;
        int sum = 0;

        while(temp > 0) {
            count++;
            temp /= 10;
        }

        temp = num;
        while (temp > 0) {
            int digit = temp % 10;
            sum += Math.pow(digit, count);
            temp /= 10;
        }

        return sum == num;
    }
}
